sharepoint/provider/client.py

```diff
@@ -1,141 +1,126 @@
-from functools import lru_cache
-from azure.identity import ClientSecretCredential
-from flask import current_app as app
-from msgraph.core import GraphClient, APIVersion
-from urllib.parse import urlparse
+import requests
+
+from msal import ConfidentialClientApplication
+from flask import current_app as app, request
 
 from . import UpstreamProviderError
-from .consts import CACHE_SIZE
 
-client = None
+AUTHORIZATION_HEADER = "Authorization"
+BEARER_PREFIX = "Bearer "
 
 
 class SharepointClient:
     DEFAULT_SCOPES = ["https://graph.microsoft.com/.default"]
     DEFAULT_REGION = "NAM"
-    SEARCH_ENTITY_TYPES = ["driveItem", "listItem"]
-    SEARCH_URL = "/search/query"
-    SEARCH_LIMIT = 3
+    BASE_URL = "https://graph.microsoft.com/v1.0"
+    SEARCH_ENTITY_TYPES = ["driveItem"]
+    APPLICATION_AUTH = "application"
+    DELEGATED_AUTH = "user"
+
+    def __init__(self, auth_type, search_limit):
+        self.access_token = None
+        self.user = None
+        self.auth_type = auth_type
+        self.search_limit = search_limit
 
-    graph_client = None
+    def get_auth_type(self):
+        return self.auth_type
 
-    def __init__(self, tenant_id, client_id, client_secret, search_limit=5):
+    def set_app_access_token(self, tenant_id, client_id, client_secret):
         try:
-            credential = ClientSecretCredential(
-                tenant_id,
-                client_id,
-                client_secret,
+            credential = ConfidentialClientApplication(
+                client_id=client_id,
+                client_credential=client_secret,
+                authority=f"https://login.microsoftonline.com/{tenant_id}",
             )
 
-            self.graph_client = GraphClient(
-                credential=credential,
+            token_response = credential.acquire_token_for_client(
                 scopes=self.DEFAULT_SCOPES,
-                api_version=APIVersion.beta,
             )
+            if "access_token" not in token_response:
+                raise UpstreamProviderError(
+                    "Error while retrieving access token from Microsoft Graph API"
+                )
+            self.access_token = token_response["access_token"]
         except Exception as e:
             raise UpstreamProviderError(
-                f"Error while initializing Sharepoint client: {str(e)}"
+                f"Error while initializing Teams client: {str(e)}"
             )
 
-        self.search_limit = search_limit
+    def set_user_access_token(self, token):
+        self.access_token = token
+        self.headers = {"Authorization": f"Bearer {self.access_token}"}
 
```

Comment:
self.headers gets set here, but in set_app_access_token() it doesn't, can we init it in the __init__ and add that to the app auth and then everywhere the headers are needed you can use self.headers instead?

Reply:
Done